c7n/resources/rds.py

```diff
@@ -281,6 +300,56 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+    """ Scan DB instances for available engine upgrades
+
+        This will pull DB instances & check their specific engine for any
+        engine version with higher release numbers than the current one
+
+        This will also tag the rds instance with 'target_engine' which is
+        the most recent version of the engine available
+    """
+
+    schema = type_schema('upgrade-available', value={'type': 'boolean'})
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+        results = []
+        engine_upgrades = _list_engines_upgrade_version(client)
+        for r in resources:
+            upgrades = engine_upgrades[r['Engine']]
+            if len(upgrades) == 0 or r['EngineVersion'] not in upgrades:
+                continue
+            target_upgrade = upgrades[r['EngineVersion']]
+            r['c7n.rds-minor-engine-upgrade'] = target_upgrade
+            if self.data.get('value', True):
+                if target_upgrade != r['EngineVersion']:
+                    if not r in results:
+                        results.append(r)
+            else:
+                if target_upgrade == r['EngineVersion']:
+                    if not r in results:
+                        results.append(r)
+        return results
+
+
+@actions.register('upgrade-minor')
+class UpgradeMinorRDS(BaseAction):
+
+    schema = type_schema(
+        'upgrade-minor', immediate={'type': 'boolean'})
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('rds')
+
+        for r in resources:
+            client.modify_db_instance(
+                DBInstanceIdentifier=r['DBInstanceIdentifier'],
+                EngineVersion=r['c7n.rds-minor-engine-upgrade'],
```

Comment:
there's an implicit dependency here on the annotation from a previous filter, without which this fails at runtime, which isn't good behavior. i think we want to use the annotation if it exists, else... also there's a question of how an applied upgrade looks like via the api, ie. if there's an applied upgrade pending or in progress, does running this twice also have a runtime error.